jks/jks.py

```diff
@@ -125,10 +125,10 @@ def _read_data(data, pos):
 
 def _sun_jks_pkey_decrypt(data, password):
     'implements private key crypto algorithm used by JKS files'
-    password = ''.join([b'\0'+c.encode('latin-1') for c in password]) # the JKS algorithm uses a regular Java UTF16-BE string for the password, so insert 0 bytes
+    password = password.encode('UTF-16BE') # the JKS algorithm uses a regular Java UTF16-BE string for the password, so insert 0 bytes
     iv, data, check = data[:20], data[20:-20], data[-20:]
     xoring = zip(data, _jks_keystream(iv, password))
-    key = ''.join([chr(ord(a) ^ ord(b)) for a, b in xoring])
+    key = bytes([a ^ b for a, b in xoring])
```

Comment:
Yea, I think this is because in Python3 a "bytes" object is like a bytearray.  So I guess ord will fail in Python 3, but ^ will fail in Python 2.

This is one of the really boneheaded (IMO) changes.  It leads to behavior like bytes(3) == b'3' in python 2, but bytes(3) == b'\0\0\0' in python 3.

We may just need to add a branch here and have a different behavior in 2 vs 3.  That is the only thing that jumped out at me; overall great work appreciate your help.